ci/ray_ci/doc/autodoc.py

```diff
@@ -0,0 +1,110 @@
+import os
+import re
+from typing import List
+
+from ci.ray_ci.doc.api import (
+    API,
+    SPHINX_AUTOSUMMARY_HEADER,
+    SPHINX_AUTOCLASS_HEADER,
+    SPHINX_CURRENTMODULE_HEADER,
+)
+
+
+class Autodoc:
+    """
+    Autodoc class represents the top level sphinx autodoc landing page and finds
+    autodoc APIs that would be generated from sphinx from all sub-pages
+    """
+
+    def __init__(self, head_rst_file: str):
+        """
+        Args:
+            head_rst_file: The path to the landing page rst file that contains the list
+            of children rsts of the autodoc APIs
+        """
+        self._head_rst_file = head_rst_file
+        self._autodoc_rsts = []
+        self._apis = []
+
+    def get_apis(self) -> List[API]:
+        self.walk()
+        return self._apis
+
+    def walk(self) -> None:
+        if self._apis:
+            # already walk
+            return
+        rsts = self._get_autodoc_rsts()
+        for rst in rsts:
+            self._apis.extend(self._parse_autodoc_rst(rst))
+
+    def _get_autodoc_rsts(self) -> List[str]:
+        """
+        Parse the list of rst declared in the head_rst_file, for exampe:
+
+        .. toctree::
+
+            area_01.rst
+            area_02.rst
+        """
+        if self._autodoc_rsts:
```

Comment:
maybe set `_autodoc_rsts` to `None` first, and check `is None` here?

it seems that this function is trying to implement "parse once and save the result for future use", but if their are no line in the file that has `.rst` suffix, this will parse the file on each call?